File: LeetCode/Miscellaneous/C++/7. Reverse Integer.cpp
#include <bits/stdc++.h>
using namespace std;

class Solution
{
public:
    int reverse(int x)
    {
        int rev = 0;

        while (x != 0)
        {
            int remainder = x % 10;

            if (rev > INT_MAX / 10 || (rev == INT_MAX / 10 && remainder > 7))
                return 0;
            if (rev < INT_MIN / 10 || (rev == INT_MIN / 10 && remainder < -8))
                return 0;

            rev = (rev * 10) + remainder;

            x /= 10;
        }

        return rev;
    }
};

File: LeetCode/Miscellaneous/Python/7. Reverse Integer.py
class Solution:
    def reverse(self, x: int) -> int:
        rev_num = 0
        sign = -1 if x < 0 else 1
        x = abs(x)
        
        while x > 0:
            rev_num = rev_num * 10 + x % 10
            x //= 10
        
        rev_num *= sign
        
        if rev_num < -2**31 or rev_num > 2**31 - 1:
            return 0
        
        return rev_num
